359.py

```diff
@@ -0,0 +1,23 @@
+class Logger:
+
+    def __init__(self):
+        """
+        Initialize your data structure here.
+        """
+        self.messages = collections.defaultdict(str)
```

Comment:
Two issues here:

1. Your implementation of `shouldPrintMessage` does not have a use case where a default value is needed from `self.messages`, so there is no need to create a `defaultdict` here.
2. The syntax of `defaultdict(some_type)` means that the values in your dictionary have a default value of type `some_type`. However, in your `shouldPrintMessage` function, the keys are of `str` type while the values are of `int` type. This dictionary definition is not consistent with its actual usages.

Reply:
Another bad thing about this design is, it is not scalable at all, the space will keep increasing, I will need to pop out the old records.